[CI] InternalCategorizationAggregationTests testReduceRandom failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/zgehjep7ea33k/tests/:x-pack:plugin:ml:test/org.elasticsearch.xpack.ml.aggs.categorization.InternalCategorizationAggregationTests/testReduceRandom

**Reproduction line:**
`./gradlew ':x-pack:plugin:ml:test' --tests "org.elasticsearch.xpack.ml.aggs.categorization.InternalCategorizationAggregationTests.testReduceRandom" -Dtests.seed=1102F1F32E3E5113 -Dtests.locale=en-PH -Dtests.timezone=America/Fort_Nelson -Druntime.java=17`

**Applicable branches:**
8.3

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.ml.aggs.categorization.InternalCategorizationAggregationTests&tests.test=testReduceRandom

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <{BytmNQmzGpd LwDLQwia LwDLQwia=10, LQTbOOutrDfOyljaYs GCEme ecWwaQXJFswod aVEmnZMiU msPJoLMOFMfuCPwDIR PnlxQcnSIgmoY ecWwaQXJFswod msPJoLMOFMfuCPwDIR GCEme=10, kGrSWON sMeugtjLIvycCw zgeHWVi YuladfeJmbvOsWXbn wKlas xvdaD zgeHWVi kGrSWON kGrSWON zgeHWVi=10, VOFMdQzPwTiZbLshCkF KBn DijztwckS pkAzDyhyQaHzNGfzBfsz wwzJeygChlnTk pkAzDyhyQaHzNGfzBfsz DijztwckS=10, YiymAIlny msa vQXsumQrBJzBuF JcqcuIV HwfHGINJsFXCDGUTRe XqAlnHKLfKDaiT oYeLwEJPO msa JcqcuIV JcqcuIV HwfHGINJsFXCDGUTRe vQXsumQrBJzBuF vQXsumQrBJzBuF JcqcuIV=10, ArEnuSddrlEQwUxyGyh BpxKdpb isMNmUf epGqlHfELaetB NTHjemEQ wzNlDtFoMsxbgnYm NTHjemEQ isMNmUf isMNmUf wzNlDtFoMsxbgnYm=10, LgvmOCuJFxTOZyysBJN pKMhFuou feXTpEsJUTeadAE RSjdQOAKNVjlhO OLfc qxJIaUn GfodnV DGrCGUTsyobdvFPiTW gZHU xRcQUK nwI EPQTcOWZwMiVFcAQYfR aMMFUOQDhtCGrWSbxzlX qxJIaUn=10, cgufNCyyIoriiWrLPMA kjEDeNCDTY aovPIKGpZEWzei YRsMIXoKV rMnaxLrrboctKVsEn xpBSasaNL FhoYDyEEIBJcflo tEV kjEDeNCDTY FhoYDyEEIBJcflo aovPIKGpZEWzei FhoYDyEEIBJcflo aovPIKGpZEWzei xpBSasaNL aovPIKGpZEWzei=10, gzVXepwvHVGxHRMGcgZ QuPJEZNKQmxly YYnfAYXaSFKE gkqCnSTagRShGgCxojw NCeNzMmI pyoZGW ZdWtXezO ZdWtXezO gzVXepwvHVGxHRMGcgZ YYnfAYXaSFKE YYnfAYXaSFKE gzVXepwvHVGxHRMGcgZ YYnfAYXaSFKE gkqCnSTagRShGgCxojw=10, bBMozwiGWkn VhfxDplYrdkpUjn VhfxDplYrdkpUjn=10, iLrJoDN EJzYsALoHygqxa HaYDKjOTVoo nbvmLBe OIVUmaXvpUcBEI ESVyqYuMpmgD ntMqivVBmF rsAFvh Ptz AKCLJUOAiNAps AKCLJUOAiNAps EJzYsALoHygqxa Ptz=10, XLyXoiyzoTlDBieWAhuA cypAWiV zuxHOgWvwmzelU plgbvFVytnDMpj JbeoVH IPKgPKtJYeDNIlUcM zQVlndcoRd lbBclSdGymLowRNIX XLyXoiyzoTlDBieWAhuA cypAWiV zuxHOgWvwmzelU XLyXoiyzoTlDBieWAhuA cypAWiV=10, EvRJn DhzxZKhUIuTBOEG ynsYpDhyfFJIR UlqnOvfESEXVIUpZAWl FLAScLAqzve srFPrSMrebu HFT ahGNfmX jBYGFxMmzelZLuW DZbWaloQnCdzkYG HFT DZbWaloQnCdzkYG UlqnOvfESEXVIUpZAWl DZbWaloQnCdzkYG=10, BjZDCNSqIjHBqPDH NkPOdtlh QqFKaZgjpIoOVViW csUULPZ fndBYDd JWFPUH CABxoIIAcIM jYqpE udQqodjbAOuFN QSpUsgZESuXPQEWjnROO NkPOdtlh JWFPUH jYqpE fndBYDd csUULPZ=10, jQtchwRVoc dYTJvZigwJXKYJLZMj oNuJUaAwbfu cCPkeAs dOuyTOWwAN qFtYbS ywISSEZeXarlOTRzMX qFtYbS ywISSEZeXarlOTRzMX ywISSEZeXarlOTRzMX ywISSEZeXarlOTRzMX=10}>
     but: was <{LgvmOCuJFxTOZyysBJN pKMhFuou feXTpEsJUTeadAE RSjdQOAKNVjlhO OLfc qxJIaUn GfodnV DGrCGUTsyobdvFPiTW gZHU xRcQUK nwI EPQTcOWZwMiVFcAQYfR aMMFUOQDhtCGrWSbxzlX qxJIaUn=10, cnA cnA cnA cnA=16, EvRJn DhzxZKhUIuTBOEG ynsYpDhyfFJIR UlqnOvfESEXVIUpZAWl FLAScLAqzve srFPrSMrebu HFT ahGNfmX jBYGFxMmzelZLuW DZbWaloQnCdzkYG HFT DZbWaloQnCdzkYG UlqnOvfESEXVIUpZAWl DZbWaloQnCdzkYG=10, BytmNQmzGpd LwDLQwia LwDLQwia=10, LQTbOOutrDfOyljaYs GCEme ecWwaQXJFswod aVEmnZMiU msPJoLMOFMfuCPwDIR PnlxQcnSIgmoY ecWwaQXJFswod msPJoLMOFMfuCPwDIR GCEme=10, bBMozwiGWkn VhfxDplYrdkpUjn VhfxDplYrdkpUjn=10, jQtchwRVoc dYTJvZigwJXKYJLZMj oNuJUaAwbfu cCPkeAs dOuyTOWwAN qFtYbS ywISSEZeXarlOTRzMX qFtYbS ywISSEZeXarlOTRzMX ywISSEZeXarlOTRzMX ywISSEZeXarlOTRzMX=10, iLrJoDN EJzYsALoHygqxa HaYDKjOTVoo nbvmLBe OIVUmaXvpUcBEI ESVyqYuMpmgD ntMqivVBmF rsAFvh Ptz AKCLJUOAiNAps AKCLJUOAiNAps EJzYsALoHygqxa Ptz=10, cgufNCyyIoriiWrLPMA kjEDeNCDTY aovPIKGpZEWzei YRsMIXoKV rMnaxLrrboctKVsEn xpBSasaNL FhoYDyEEIBJcflo tEV kjEDeNCDTY FhoYDyEEIBJcflo aovPIKGpZEWzei FhoYDyEEIBJcflo aovPIKGpZEWzei xpBSasaNL aovPIKGpZEWzei=10, kGrSWON sMeugtjLIvycCw zgeHWVi YuladfeJmbvOsWXbn wKlas xvdaD zgeHWVi kGrSWON kGrSWON zgeHWVi=10, XLyXoiyzoTlDBieWAhuA cypAWiV zuxHOgWvwmzelU plgbvFVytnDMpj JbeoVH IPKgPKtJYeDNIlUcM zQVlndcoRd lbBclSdGymLowRNIX XLyXoiyzoTlDBieWAhuA cypAWiV zuxHOgWvwmzelU XLyXoiyzoTlDBieWAhuA cypAWiV=10, gzVXepwvHVGxHRMGcgZ QuPJEZNKQmxly YYnfAYXaSFKE gkqCnSTagRShGgCxojw NCeNzMmI pyoZGW ZdWtXezO ZdWtXezO gzVXepwvHVGxHRMGcgZ YYnfAYXaSFKE YYnfAYXaSFKE gzVXepwvHVGxHRMGcgZ YYnfAYXaSFKE gkqCnSTagRShGgCxojw=10, VOFMdQzPwTiZbLshCkF KBn DijztwckS pkAzDyhyQaHzNGfzBfsz wwzJeygChlnTk pkAzDyhyQaHzNGfzBfsz DijztwckS=10, BjZDCNSqIjHBqPDH NkPOdtlh QqFKaZgjpIoOVViW csUULPZ fndBYDd JWFPUH CABxoIIAcIM jYqpE udQqodjbAOuFN QSpUsgZESuXPQEWjnROO NkPOdtlh JWFPUH jYqpE fndBYDd csUULPZ=10, YiymAIlny msa vQXsumQrBJzBuF JcqcuIV HwfHGINJsFXCDGUTRe XqAlnHKLfKDaiT oYeLwEJPO msa JcqcuIV JcqcuIV HwfHGINJsFXCDGUTRe vQXsumQrBJzBuF vQXsumQrBJzBuF JcqcuIV=10, ArEnuSddrlEQwUxyGyh BpxKdpb isMNmUf epGqlHfELaetB NTHjemEQ wzNlDtFoMsxbgnYm NTHjemEQ isMNmUf isMNmUf wzNlDtFoMsxbgnYm=10}>

  at __randomizedtesting.SeedInfo.seed([1102F1F32E3E5113:D13F5DE20863C845]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.ml.aggs.categorization.InternalCategorizationAggregationTests.assertReduced(InternalCategorizationAggregationTests.java:74)
  at org.elasticsearch.xpack.ml.aggs.categorization.InternalCategorizationAggregationTests.assertReduced(InternalCategorizationAggregationTests.java:36)
  at org.elasticsearch.test.InternalAggregationTestCase.testReduceRandom(InternalAggregationTestCase.java:475)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/aggs/categorization/InternalCategorizationAggregationTests.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/aggs/categorization/InternalCategorizationAggregationTests.java
@@ -46,12 +46,6 @@ public class InternalCategorizationAggregationTests extends InternalMultiBucketA
         return MachineLearningTests.createTrialLicensedMachineLearning(Settings.EMPTY);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/87240")
-    public void testReduceRandom() {
-        // The bug is in the assertReduced() method immediately below that the base class testReduceRandom() calls.
-        // To unmute after the bug is fixed, simply delete this entire method so that the base class method is used again.
-    }
-
     @Override
     protected void assertReduced(InternalCategorizationAggregation reduced, List<InternalCategorizationAggregation> inputs) {
         Map<Object, Long> reducedCounts = toCounts(reduced.getBuckets().stream());
@@ -80,7 +74,7 @@ public class InternalCategorizationAggregationTests extends InternalMultiBucketA
             name,
             randomIntBetween(10, 100),
             randomLongBetween(1, 10),
-            randomIntBetween(1, 100),
+            randomIntBetween(50, 100),
             metadata,
             buckets
         );

